<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Retrieve the differential expression results for a given gene among datasets matching the provided query and filter — get_gene_differential_expression_values • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Retrieve the differential expression results for a given gene among datasets matching the provided query and filter — get_gene_differential_expression_values"><meta property="og:description" content="Retrieve the differential expression results for a given gene among datasets matching the provided query and filter"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Retrieve the differential expression results for a given gene among datasets matching the provided query and filter</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/devel/R/allEndpoints.R" class="external-link"><code>R/allEndpoints.R</code></a></small>
    <div class="hidden name"><code>get_gene_differential_expression_values.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieve the differential expression results for a given gene among datasets matching the provided query and filter</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_gene_differential_expression_values</span><span class="op">(</span></span>
<span>  <span class="va">gene</span>,</span>
<span>  query <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  taxa <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  uris <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  filter <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.raw"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>  file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.file"</span>, <span class="cn">NA_character_</span><span class="op">)</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.overwrite"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-gene">gene<a class="anchor" aria-label="anchor" href="#arg-gene"></a></dt>
<dd><p>An ensembl gene identifier which typically starts with ensg or an ncbi gene identifier or an official gene symbol approved by hgnc</p></dd>


<dt id="arg-query">query<a class="anchor" aria-label="anchor" href="#arg-query"></a></dt>
<dd><p>The search query. Queries can include plain text or ontology
terms They also support conjunctions ("alpha AND beta"), disjunctions ("alpha OR beta")
grouping ("(alpha OR beta) AND gamma"), prefixing ("alpha*"), wildcard characters
("BRCA?") and fuzzy matches ("alpha~").</p></dd>


<dt id="arg-taxa">taxa<a class="anchor" aria-label="anchor" href="#arg-taxa"></a></dt>
<dd><p>A vector of taxon common names (e.g. human, mouse, rat). Providing multiple
species will return results for all species. These are appended
to the filter and equivalent to filtering for <code>taxon.commonName</code> property</p></dd>


<dt id="arg-uris">uris<a class="anchor" aria-label="anchor" href="#arg-uris"></a></dt>
<dd><p>A vector of ontology term URIs. Providing multiple terms will
return results containing any of the terms and their children. These are
appended to the filter and equivalent to filtering for <code>allCharacteristics.valueUri</code></p></dd>


<dt id="arg-filter">filter<a class="anchor" aria-label="anchor" href="#arg-filter"></a></dt>
<dd><p>Filter results by matching expression. Use <code><a href="filter_properties.html">filter_properties</a></code>
function to get a list of all available parameters. These properties can be
combined using "and" "or" clauses and may contain common operators such as "=", "&lt;" or "in".
(e.g. "taxon.commonName = human", "taxon.commonName in (human,mouse), "id &lt; 1000")</p></dd>


<dt id="arg-threshold">threshold<a class="anchor" aria-label="anchor" href="#arg-threshold"></a></dt>
<dd><p>number</p></dd>


<dt id="arg-raw">raw<a class="anchor" aria-label="anchor" href="#arg-raw"></a></dt>
<dd><p><code>TRUE</code> to receive results as-is from Gemma, or <code>FALSE</code> to enable
parsing. Raw results usually contain additional fields and flags that are
omitted in the parsed results.</p></dd>


<dt id="arg-memoised">memoised<a class="anchor" aria-label="anchor" href="#arg-memoised"></a></dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>


<dt id="arg-file">file<a class="anchor" aria-label="anchor" href="#arg-file"></a></dt>
<dd><p>The name of a file to save the results to, or <code>NULL</code> to not write
results to a file. If <code>raw == TRUE</code>, the output will be the raw endpoint from the
API, likely a JSON or a gzip file. Otherwise, it will be a RDS file.</p></dd>


<dt id="arg-overwrite">overwrite<a class="anchor" aria-label="anchor" href="#arg-overwrite"></a></dt>
<dd><p>Whether or not to overwrite if a file exists at the specified
filename.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data.table containing differential expression results. This table
is stripped down some relevant information for speed of execution. Details about
the contrasts can be accessesed via <code><a href="get_result_sets.html">get_result_sets</a></code> function</p>
<p>The fields of the output data.table are:</p>
<ul><li><p><code>result.ID</code>: Result set ID of the differential expression analysis.
May represent multiple factors in a single model.</p></li>
<li><p><code>contrast.ID</code>: Id of the specific contrast factor. Together with the result.ID
they uniquely represent a given contrast.</p></li>
<li><p><code>experiment.ID</code>: Id of the source experiment</p></li>
<li><p><code>factor.coefficient</code>: Model coefficient calculated for the specific contrast factor</p></li>
<li><p><code>factor.logfc</code>: Log 2 fold change calculated for the specific contrast factor</p></li>
<li><p><code>factor.pvalue</code>: p values calculated for the specific contrast factor</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># get all differential expression results for ENO2</span></span></span>
<span class="r-in"><span><span class="co"># from datasets marked with the ontology term for brain</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">get_gene_differential_expression_values</span><span class="op">(</span><span class="fl">2026</span>, uris <span class="op">=</span> <span class="st">"http://purl.obolibrary.org/obo/UBERON_0000955"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    result.ID contrast.ID experiment.ID pvalue correctedPValue   rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        &lt;int&gt;      &lt;char&gt;         &lt;int&gt;  &lt;num&gt;           &lt;num&gt;  &lt;num&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:    565737         118            18 0.4213          0.8083 0.5213</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:    560596      135393           134 0.5214          0.9999 0.4142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:    560598        1060           134 0.0450          0.5422 0.0829</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:    560599        1060           134 0.1648          0.9985 0.1480</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:    560738       15628           233 0.1751          0.4526 0.3869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:    560738       15629           233 0.1751          0.4526 0.3869</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    factor.coefficient factor.logfc factor.pvalue baseline.ID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 &lt;num&gt;        &lt;num&gt;         &lt;num&gt;      &lt;char&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:            -0.1185      -0.1185        0.4213         119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2:            -0.2146      -0.2146        0.5300      135392</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3:             0.4842       0.4842        0.2025        1059</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4:             0.4842       0.4842        0.1648        1059</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5:            -0.1005      -0.1005        0.6375      117789</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6:             0.2426       0.2426        0.1294      117789</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

