<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compile gene expression data and metadata — get_dataset_object • gemma.R</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compile gene expression data and metadata — get_dataset_object"><meta property="og:description" content="Return an annotated Bioconductor-compatible
data structure or a long form tibble of the queried dataset, including
expression data and the experimental design."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compile gene expression data and metadata</h1>
    <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/devel/R/convenience.R" class="external-link"><code>R/convenience.R</code></a></small>
    <div class="hidden name"><code>get_dataset_object.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Return an annotated Bioconductor-compatible
data structure or a long form tibble of the queried dataset, including
expression data and the experimental design.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_dataset_object</span><span class="op">(</span></span>
<span>  <span class="va">datasets</span>,</span>
<span>  genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keepNonSpecific <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  consolidate <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  resultSets <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  contrasts <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  metaType <span class="op">=</span> <span class="st">"text"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  memoised <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">getOption</a></span><span class="op">(</span><span class="st">"gemma.memoised"</span>, <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-datasets">datasets<a class="anchor" aria-label="anchor" href="#arg-datasets"></a></dt>
<dd><p>A vector of dataset IDs or short names</p></dd>


<dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p>A vector of NCBI IDs, Ensembl IDs or gene symbols.</p></dd>


<dt id="arg-keepnonspecific">keepNonSpecific<a class="anchor" aria-label="anchor" href="#arg-keepnonspecific"></a></dt>
<dd><p>logical. <code>FALSE</code> by default. If <code>TRUE</code>, results
from probesets that are not specific to the gene will also be returned.</p></dd>


<dt id="arg-consolidate">consolidate<a class="anchor" aria-label="anchor" href="#arg-consolidate"></a></dt>
<dd><p>An option for gene expression level consolidation. If empty,
will return every probe for the genes. "pickmax" to
pick the probe with the highest expression, "pickvar" to pick the prove with
the highest variance and "average" for returning the average expression</p></dd>


<dt id="arg-resultsets">resultSets<a class="anchor" aria-label="anchor" href="#arg-resultsets"></a></dt>
<dd><p>Result set IDs of the a differential expression analysis. Optional. If provided, the output will only include
the samples from the subset used in the result set ID.
Must be the same length as <code>datasets</code>.'</p></dd>


<dt id="arg-contrasts">contrasts<a class="anchor" aria-label="anchor" href="#arg-contrasts"></a></dt>
<dd><p>Contrast IDs of a differential expression contrast. Optional. Need resultSets to be defined to work. If provided, the
output will only include samples relevant to the specific contrats.</p></dd>


<dt id="arg-metatype">metaType<a class="anchor" aria-label="anchor" href="#arg-metatype"></a></dt>
<dd><p>How should the metadata information should be included. Can be "text", "uri" or "both". "text" and "uri" options</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>"se"for a SummarizedExperiment or "eset" for Expression Set. We recommend using
SummarizedExperiments which are more recent. See the Summarized experiment
<a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">vignette</a>
or the ExpressionSet <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link">vignette</a>
for more details. "tidy" for a long form data frame compatible with tidyverse functions.
'list' to return a list containing individual data frames containing expression values,
design and the experiment.</p></dd>


<dt id="arg-memoised">memoised<a class="anchor" aria-label="anchor" href="#arg-memoised"></a></dt>
<dd><p>Whether or not to save to cache for future calls with the
same inputs and use the result saved in cache if a result is already saved.
Doing <code>options(gemma.memoised = TRUE)</code> will ensure that the cache is always
used. Use <code><a href="forget_gemma_memoised.html">forget_gemma_memoised</a></code> to clear the cache.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list of <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code>s,
<code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">ExpressionSet</a></code>s or a tibble containing metadata and
expression data for the queried datasets and genes. Metadata will be expanded to include
a variable number of factors that annotates samples from a dataset but will
always include single "factorValues" column that houses data.tables that
include all annotations for a given sample.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu">get_dataset_object</span><span class="op">(</span><span class="st">"GSE2018"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $`1`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: SummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 18006 34 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(8): title abstract ... gemmaSuitabilityScore taxon</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(18006): 1007_s_at 1053_at ... AFFX-HUMISGF3A/M97935_MA_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   AFFX-HUMISGF3A/M97935_MB_at</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(4): Probe GeneSymbol GeneName NCBIid</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(34): BAL_47c_A0B1 BAL_47b_A0B1 ... BAL_15a_A0B1 BAL_37_A1B1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(3): factorValues block disease</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

