<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing curated expression data with gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing curated expression data with gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated expression data with gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gemma.R_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing curated expression data with gemma.R</h1>
                        <h4 data-toc-skip class="author">Javier Castillo-Arnemann</h4>
            
            <h4 data-toc-skip class="date">February 18, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/gemma.R.Rmd" class="external-link"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="about-gemma">About Gemma<a class="anchor" aria-label="anchor" href="#about-gemma"></a>
</h2>
<p><a href="https://gemma.ubc.msl.ca/" class="external-link">Gemma</a> is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, check this <a href="https://pavlidislab.github.io/Gemma/curation.html" class="external-link">page</a> or the latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045" class="external-link">publication</a>.</p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<div class="section level3">
<h3 id="development">Development<a class="anchor" aria-label="anchor" href="#development"></a>
</h3>
<p>The current development version of <code>gemma.R</code> can be installed with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"PavlidisLab/gemma.R"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bioconductor-not-available-yet">Bioconductor (not available yet)<a class="anchor" aria-label="anchor" href="#bioconductor-not-available-yet"></a>
</h3>
<p>You can install <code>gemma.R</code> through <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> with the following code:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"gemma.R"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="downloading-expression-data">Downloading expression data<a class="anchor" aria-label="anchor" href="#downloading-expression-data"></a>
</h2>
<p>The main goal of this wrapper is to enable easy access to Gemma’s curated datasets for downstream analyses or meta-analyses combining multiple datasets. In this example, we want to find datasets that are associated with bipolar disorder, and we are only interested in human data. In addition, we’ll subset our results to datasets that have been batch corrected.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/searchDatasets.html">searchDatasets</a></span><span class="op">(</span><span class="st">"bipolar"</span>, limit <span class="op">=</span> <span class="fl">100</span>, taxon <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">geeq.batchCorrected</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Accession</span>, <span class="va">ee.Database</span>, <span class="va">ee.Samples</span><span class="op">)</span></code></pre></div>
<pre><code>   ee.ShortName                                                         ee.Name
1:     GSE35974                 Expression data from the human cerebellum brain
2:     GSE46416 State- and trait-specific gene expression in euthymia and mania
   ee.ID ee.Accession ee.Database ee.Samples
1:  5939     GSE35974         GEO        144
2:  8997     GSE46416         GEO         32</code></pre>
<p>We are left with two datasets, and we’ll pick <a href="https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=5939" class="external-link">GSE35974</a> since it has the larger number of samples. Now that we have the ID for our experiment, we can fetch the data associated with it.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetsInfo.html">getDatasetsInfo</a></span><span class="op">(</span><span class="st">"GSE35974"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Description</span><span class="op">)</span></code></pre></div>
<pre><code>   ee.ShortName                                         ee.Name ee.ID
1:     GSE35974 Expression data from the human cerebellum brain  5939
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            ee.Description
1: Background: Schizophrenia (SCZ) and bipolar disorder (BD) are highly heritable psychiatric disorders. Associated genetic and gene expression changes have been identified, but many have not been replicated and have unknown functions. We identified groups of genes whose expressions varied together, that is co-expression modules, then tested them for association with SCZ. Using weighted gene co-expression network analysis, we show that two modules were differentially expressed in patients versus controls. One, upregulated in cerebral cortex, was enriched with neuron differentiation and neuron development genes, as well as disease genome-wide association study genetic signals; the second, altered in cerebral cortex and cerebellum, was enriched with genes involved in neuron protection functions. The findings were preserved in five expression data sets, including sets from three brain\nregions, from a different microarray platform, and from BD patients. From those observations, we propose neuron differentiation and development pathways may be involved in etiologies of both SCZ and BD, and neuron protection function participates in pathological process of the diseases.\nLast Updated (by provider): Jul 26, 2018\nContributors:  Chao Chen Chunyu Liu Lijun Cheng</code></pre>
<p>To access the expression data in a convenient form, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getDataset.html" class="external-link"><code>getDataset()</code></a>. It is a high-level wrapper that combines various endpoints to return an annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link"><code>ExpressionSet</code></a> of the queried dataset that is compatible with other Bioconductor packages for downstream analyses. They include the expression matrix along with the experimental design, and ensure the sample names match between both when transforming/subsetting data.</p>
<p>Alternatively, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetExpression.html" class="external-link"><code>getDatasetExpression()</code></a> to only get the expression matrix, and <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetDesign.html" class="external-link"><code>getDatasetDesign()</code></a> to get the experimental design matrix. Finally, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getTidyDataset.html" class="external-link"><code>getTidyDataset()</code></a> to get a tibble that combines the expression and design matrices in tidy format. This format is less memory efficient than <a href="https://pavlidislab.github.io/gemma.R/reference/getDataset.html" class="external-link"><code>getDataset()</code></a> but allows easy visualization and exploration with <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> and the rest of the <a href="https://www.tidyverse.org/" class="external-link">tidyverse</a>.</p>
<p>Since this dataset also contains patients with schizophrenia and depression, and we’re only interested in bipolar patients, we can remove those samples. We’re also keeping the <code>reference_subject_role</code>s, which refers to the control samples in Gemma datasets.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDataset.html">getDataset</a></span><span class="op">(</span><span class="st">"GSE35974"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Check the levels of the DiseaseState cofactor</span>
<span class="va">dat</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>[1] "reference_subject_role" "mental_depression"      "Bipolar_Disorder"      
[4] "schizophrenia"         </code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset patients with bipolar disorder and controls.</span>
<span class="va">bipolar</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[</span>, <span class="va">dat</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op">==</span> <span class="st">"Bipolar_Disorder"</span> <span class="op">|</span> <span class="va">dat</span><span class="op">$</span><span class="va">DiseaseState</span> <span class="op">==</span> <span class="st">"reference_subject_role"</span><span class="op">]</span>
<span class="va">bipolar</span></code></pre></div>
<pre><code>class: SummarizedExperiment 
dim: 21939 87 
metadata(8): title abstract ... GemmaSuitabilityScore taxon
assays(1): counts
rownames(21939): 7896736 7896742 ... 8180166 8180179
rowData names(3): GeneSymbol GeneName NCBIid
colnames(87): DE59_111109_rep1 DE7_111109 ... AC103_100909 AC1_100909
colData names(2): DiseaseState batch</code></pre>
<p>Let’s check the expression for every sample to make sure they look OK:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get Expression matrix</span>
<span class="va">bipExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">bipolar</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">bipExpr</span>, pch <span class="op">=</span> <span class="st">"."</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, xlab <span class="op">=</span> <span class="st">"Sample"</span>, ylab <span class="op">=</span> <span class="st">"Expression"</span><span class="op">)</span></code></pre></div>
<p><img src="gemma.R_files/figure-html/boxplot-1.png" width="700"></p>
<p>We can quickly see three samples among bipolar patients and controls are empty. This is because during the curation process Gemma flags potential outlier samples automatically, which are then manually reviewed. Samples confirmed to be outliers are represented as missing data to ensure transparency.</p>
<div class="section level3">
<h3 id="differential-expression-results">Differential expression results<a class="anchor" aria-label="anchor" href="#differential-expression-results"></a>
</h3>
<p>Gemma also contains differential expression results for most of its datasets, and some datasets contain more than one analysis to account for different factors and their interactions. These tables are stored as resultSets, and you can access them using <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetDE.html" class="external-link"><code>getDatasetDE()</code></a>. From here on, you can visualize/explore the data and find the most differentially-expressed genes:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDatasetDE.html">getDatasetDE</a></span><span class="op">(</span><span class="st">"GSE35974"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Classify probes for plotting</span>
<span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">&lt;-</span> <span class="st">"No"</span>
<span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_Bipolar Disorder_logFoldChange`</span> <span class="op">&gt;</span> <span class="fl">0.4</span> <span class="op">&amp;</span> 
        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_Bipolar Disorder_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Up"</span>
<span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_Bipolar Disorder_logFoldChange`</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">0.4</span> <span class="op">&amp;</span> 
        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_Bipolar Disorder_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Down"</span>

<span class="co"># Upregulated probes</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Up"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_Bipolar Disorder_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_Bipolar Disorder_pvalue`</span>, 
        <span class="va">`contrast_Bipolar Disorder_logFoldChange`</span><span class="op">)</span></code></pre></div>
<pre><code>      Probe   GeneSymbol contrast_Bipolar Disorder_pvalue
 1: 7944716                                     3.438e-06
 2: 7894738                                     5.992e-05
 3: 7893193                                     3.134e-04
 4: 7896279                                     4.714e-04
 5: 8085579                                     1.354e-03
 6: 7971661 DLEU2|MIR15A                        1.410e-03
 7: 8175531         CDR1                        1.884e-03
 8: 7892982                                     1.075e-02
 9: 8124521  H4C12|H4C11                        1.326e-02
10: 8124413         H4C4                        1.980e-02
11: 8160546       LINGO2                        3.519e-02
12: 8124385         H4C2                        3.552e-02
13: 7911241  OR2L8|OR2L3                        4.484e-02
    contrast_Bipolar Disorder_logFoldChange
 1:                                  0.4661
 2:                                  0.4625
 3:                                  0.6789
 4:                                  0.5227
 5:                                  0.4355
 6:                                  0.5317
 7:                                  0.4863
 8:                                  0.4874
 9:                                  0.4094
10:                                  0.4870
11:                                  0.5954
12:                                  0.4018
13:                                  0.4940</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Downregulated probes</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Down"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_Bipolar Disorder_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_Bipolar Disorder_pvalue`</span>, 
        <span class="va">`contrast_Bipolar Disorder_logFoldChange`</span><span class="op">)</span></code></pre></div>
<pre><code>      Probe GeneSymbol contrast_Bipolar Disorder_pvalue
 1: 8091678      VEPH1                        6.040e-06
 2: 7893969                                   9.489e-06
 3: 7892510                                   1.246e-04
 4: 7894362                                   2.389e-04
 5: 7893358                                   2.494e-04
 6: 8179481    HLA-DRA                        3.364e-04
 7: 8118548    HLA-DRA                        3.512e-04
 8: 8000097        ZP2                        5.219e-04
 9: 8178193    HLA-DRA                        6.375e-04
10: 7894590                                   1.065e-02
    contrast_Bipolar Disorder_logFoldChange
 1:                                 -0.5999
 2:                                 -0.4392
 3:                                 -0.4467
 4:                                 -0.5490
 5:                                 -0.6875
 6:                                 -0.4518
 7:                                 -0.4645
 8:                                 -0.4873
 9:                                 -0.5035
10:                                 -0.4167</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Add gene symbols as labels to DE genes</span>
<span class="va">de</span><span class="op">$</span><span class="va">delabel</span> <span class="op">&lt;-</span> <span class="st">""</span>
<span class="va">de</span><span class="op">$</span><span class="va">delabel</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">de</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span>

<span class="co"># Volcano plot for bipolar patients vs controls</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">de</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
        x <span class="op">=</span> <span class="va">`contrast_Bipolar Disorder_logFoldChange`</span>,
        y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">`contrast_Bipolar Disorder_pvalue`</span><span class="op">)</span>,
        color <span class="op">=</span> <span class="va">diffexpr</span>,
        label <span class="op">=</span> <span class="va">delabel</span>
    <span class="op">)</span>
<span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.4</span>, <span class="fl">0.4</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log2(FoldChange)"</span>, y <span class="op">=</span> <span class="st">"-log10(p-value)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="gemma.R_files/figure-html/diffExpr-1.png" width="700"></p>
<p>The unlabeled colored points in the volcano plot correspond to microarray probes that do not map to any gene after re-aligning the probes during Gemma’s curation process. You can also see how multiple probes can map to the same gene.</p>
<p>Since there are some datasets that have more than one ResultSet associated with them, you will get an error if you try to fetch that data directly:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetDE.html">getDatasetDE</a></span><span class="op">(</span><span class="st">"GSE2872"</span><span class="op">)</span></code></pre></div>
<pre><code>Error in getDatasetDE("GSE2872"): There are multiple resultSets for this dataset. Check the available resultSets with `getDatasetResultSets()` or choose all = TRUE</code></pre>
<p>In this case, you can check the available resultSets to get the ID you want and or use <code>all = TRUE</code> to get a list containing all the resultSets.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Show available resultSets</span>
<span class="fu"><a href="../reference/getDatasetResultSets.html">getDatasetResultSets</a></span><span class="op">(</span><span class="st">"GSE2872"</span><span class="op">)</span></code></pre></div>
<pre><code>   resultSet.id       factor.category                        factor.levels
1:       500185             treatment           low potassium (LK), GW5074
2:       500186             timepoint                             3 h, 6 h
3:       500187 treatment x timepoint low potassium (LK), GW5074, 3 h, 6 h</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get differential expression resultSet</span>
<span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDatasetDE.html">getDatasetDE</a></span><span class="op">(</span><span class="st">"GSE2872"</span>, resultSet <span class="op">=</span> <span class="fl">500185</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">de</span><span class="op">)</span></code></pre></div>
<pre><code>          Probe NCBIid GeneSymbol                        GeneName  pvalue
1:   1373766_at                                                   0.89790
2:   1385651_at                                                   0.02789
3:   1395741_at                                                   0.28530
4: 1384991_a_at 362981     Hdac10          histone deacetylase 10 0.29140
5:   1388337_at 290029        Pnp purine nucleoside phosphorylase 0.72850
6:   1379952_at                                                   0.53970
   corrected_pvalue    rank contrast_potassium, low potassium_logFoldChange
1:                1 0.87660                                        -0.06624
2:                1 0.01515                                         0.01429
3:                1 0.21280                                         0.16040
4:                1 0.21860                                         0.09498
5:                1 0.67500                                        -0.20260
6:                1 0.46250                                         0.01362
   contrast_potassium, low potassium_tstat
1:                                -0.19810
2:                                 0.30850
3:                                 0.86730
4:                                 0.89850
5:                                -0.68020
6:                                 0.08058
   contrast_potassium, low potassium_pvalue
1:                                   0.8479
2:                                   0.7656
3:                                   0.4110
4:                                   0.3952
5:                                   0.5156
6:                                   0.9378</code></pre>
</div>
<div class="section level3">
<h3 id="larger-queries">Larger queries<a class="anchor" aria-label="anchor" href="#larger-queries"></a>
</h3>
<p>The <code>*Info()</code> endpoints accept multiple identifiers in a single function call. For example, getting information on 2 datasets at the same time.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetsInfo.html">getDatasetsInfo</a></span><span class="op">(</span>datasets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Accession</span>, <span class="va">ee.Database</span>, <span class="va">ee.Samples</span>, <span class="va">taxon.Name</span><span class="op">)</span></code></pre></div>
<pre><code>   ee.ShortName                                                         ee.Name
1:     GSE35974                 Expression data from the human cerebellum brain
2:     GSE46416 State- and trait-specific gene expression in euthymia and mania
   ee.ID ee.Accession ee.Database ee.Samples taxon.Name
1:  5939     GSE35974         GEO        144      human
2:  8997     GSE46416         GEO         32      human</code></pre>
<p>To query large amounts of data, the API has a pagination system which uses the <code>limit</code> and <code>offset</code> arguments. To avoid overloading the system, calls are limited to a maximum of 100 entries, so the offset allows you to get the next batch of entries in the next call(s). For simplicity, this example shows how pagination works with 5 entries per query.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span>
    <span class="fu"><a href="../reference/getPlatformsInfo.html">getPlatformsInfo</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">5</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">platform.ID</span>, <span class="va">platform.ShortName</span>, <span class="va">taxon.Name</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
<pre><code>[[1]]
   platform.ID platform.ShortName taxon.Name
1:           1              GPL96      human
2:           2            GPL1355        rat
3:           3            GPL1261      mouse
4:           4             GPL570      human
5:           5              GPL81      mouse

[[2]]
   platform.ID platform.ShortName taxon.Name
1:           6              GPL85        rat
2:           7             GPL339      mouse
3:           8              GPL91      human
4:           9             GPL890        rat
5:          10            GPL1406      mouse

[[3]]
   platform.ID platform.ShortName taxon.Name
1:          11             GPL891      mouse
2:          12              GPL82      mouse
3:          13             GPL560      mouse
4:          14            GPL1073      mouse
5:          16             GPL369      mouse</code></pre>
<p>The rest of the endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE12649"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ee.shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/rbindlist.html" class="external-link">rbindlist</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.shortName</span>, <span class="va">class.Type</span>, <span class="va">class.Name</span>, <span class="va">term.Name</span><span class="op">)</span></code></pre></div>
<pre><code>    ee.shortName    class.Type     class.Name              term.Name
 1:     GSE35974   FactorValue        disease       Bipolar Disorder
 2:     GSE35974   BioMaterial biological sex                   male
 3:     GSE35974   BioMaterial biological sex                 female
 4:     GSE35974   FactorValue        disease          schizophrenia
 5:     GSE35974 ExperimentTag  organism part             cerebellum
 6:     GSE35974   FactorValue        disease      mental depression
 7:     GSE12649   BioMaterial  organism part reference subject role
 8:     GSE12649   BioMaterial        disease       Bipolar Disorder
 9:     GSE12649 ExperimentTag  organism part      prefrontal cortex
10:     GSE12649   BioMaterial        disease          schizophrenia</code></pre>
</div>
<div class="section level3">
<h3 id="memoise-data">Memoise data<a class="anchor" aria-label="anchor" href="#memoise-data"></a>
</h3>
<p>To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise" class="external-link"><code>memoise</code></a> package. To enable memoisation, simply set <code>memoised = T</code>.</p>
<p>If you’re done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you’re getting up-to-date data from Gemma, you can clear the cache using <a href="https://pavlidislab.github.io/gemma.R/reference/forgetGemmaMemoised.html" class="external-link"><code>forgetGemmaMemoised()</code></a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<div class="section level3">
<h3 id="raw-data">Raw data<a class="anchor" aria-label="anchor" href="#raw-data"></a>
</h3>
<p>By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use <code>raw=TRUE</code>. Instead of a data table, you’ll be served (typically) a list that represents the underlying JSON response.</p>
</div>
<div class="section level3">
<h3 id="file-outputs">File outputs<a class="anchor" aria-label="anchor" href="#file-outputs"></a>
</h3>
<p>Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to <code>file</code>. The extension for this file will be one of three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with <code>raw=TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested data tables</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the <code>overwrite=TRUE</code> parameter.</p>
</div>
<div class="section level3">
<h3 id="changing-defaults">Changing defaults<a class="anchor" aria-label="anchor" href="#changing-defaults"></a>
</h3>
<p>We’ve seen how to change <code>memoised=TRUE</code>, <code>overwrite=TRUE</code> and <code>raw=TRUE</code>… It’s possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. For example, if you want to memoise results by default, you can use:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.memoise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] SummarizedExperiment_1.24.0 Biobase_2.54.0             
 [3] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
 [5] IRanges_2.28.0              S4Vectors_0.32.3           
 [7] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
 [9] matrixStats_0.61.0          ggrepel_0.9.1              
[11] ggplot2_3.3.5               data.table_1.14.2          
[13] dplyr_1.0.8                 gemma.R_0.99.0             

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8             lattice_0.20-45        tidyr_1.2.0           
 [4] rprojroot_2.0.2        digest_0.6.29          utf8_1.2.2            
 [7] R6_2.5.1               evaluate_0.15          highr_0.9             
[10] httr_1.4.2             pillar_1.7.0           zlibbioc_1.40.0       
[13] rlang_1.0.1            curl_4.3.2             jquerylib_0.1.4       
[16] Matrix_1.3-4           rmarkdown_2.11         pkgdown_2.0.2         
[19] labeling_0.4.2         textshaping_0.3.6      desc_1.4.0            
[22] stringr_1.4.0          RCurl_1.98-1.6         munsell_0.5.0         
[25] DelayedArray_0.20.0    compiler_4.1.2         xfun_0.29             
[28] pkgconfig_2.0.3        systemfonts_1.0.2      htmltools_0.5.2       
[31] tidyselect_1.1.1       tibble_3.1.6           GenomeInfoDbData_1.2.7
[34] fansi_1.0.2            crayon_1.5.0           withr_2.4.3           
[37] bitops_1.0-7           grid_4.1.2             jsonlite_1.7.3        
[40] gtable_0.3.0           lifecycle_1.0.1        magrittr_2.0.2        
[43] scales_1.1.1           cli_3.2.0              stringi_1.7.6         
[46] cachem_1.0.6           farver_2.1.0           XVector_0.34.0        
[49] fs_1.5.2               bslib_0.3.1            ellipsis_0.3.2        
[52] ragg_1.1.3             generics_0.1.2         vctrs_0.3.8           
[55] tools_4.1.2            glue_1.6.1             purrr_0.3.4           
[58] fastmap_1.1.0          yaml_2.3.4             colorspace_2.0-2      
[61] memoise_2.0.1          knitr_1.37             sass_0.4.0            </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency, Paul Pavlidis.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
