<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accessing curated gene expression data with gemma.R • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accessing curated gene expression data with gemma.R">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.30</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gemma.R_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accessing curated gene expression data with gemma.R</h1>
                        <h4 data-toc-skip class="author">Javier Castillo-Arnemann</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia, Vancouver, Canada<br><h4 data-toc-skip class="author">Jordan Sicherman</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia, Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/gemma.R.Rmd" class="external-link"><code>vignettes/gemma.R.Rmd</code></a></small>
      <div class="hidden name"><code>gemma.R.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/slowkow/ggrepel" class="external-link">ggrepel</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="about-gemma">About Gemma<a class="anchor" aria-label="anchor" href="#about-gemma"></a>
</h2>
<p><a href="https://gemma.msl.ubc.ca/" class="external-link">Gemma</a> is a web site, database and a set of tools for the meta-analysis, re-use and sharing of genomics data, currently primarily targeted at the analysis of gene expression profiles. Gemma contains data from thousands of public studies, referencing thousands of published papers. Every dataset in Gemma has passed a rigorous curation process that re-annotates the expression platform at the sequence level, which allows for more consistent cross-platform comparisons and meta-analyses.</p>
<p>For detailed information on the curation process, read this <a href="https://pavlidislab.github.io/Gemma/curation.html" class="external-link">page</a> or the latest <a href="https://academic.oup.com/database/article/doi/10.1093/database/baab006/6143045" class="external-link">publication</a>.</p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor">Bioconductor<a class="anchor" aria-label="anchor" href="#bioconductor"></a>
</h3>
<p>You can install <code>gemma.R</code> through <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> with the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"gemma.R"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="downloading-expression-data">Downloading expression data<a class="anchor" aria-label="anchor" href="#downloading-expression-data"></a>
</h2>
<p>The main goal of this wrapper is to enable easy access to Gemma’s curated datasets for downstream analyses or meta-analyses combining multiple datasets. In this example, we want to find datasets that are associated with bipolar disorder, and we are only interested in human data. In addition, we’ll subset our results to datasets that have been batch corrected.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/searchDatasets.html">searchDatasets</a></span><span class="op">(</span><span class="st">"bipolar"</span>, limit <span class="op">=</span> <span class="fl">100</span>, taxon <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">geeq.batchCorrected</span> <span class="op">==</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Accession</span>, <span class="va">ee.Samples</span><span class="op">)</span></span></code></pre></div>
<pre><code>   ee.ShortName                                                         ee.Name
1:     GSE35974                 Expression data from the human cerebellum brain
2:     GSE46416 State- and trait-specific gene expression in euthymia and mania
   ee.ID ee.Accession ee.Samples
1:  5939     GSE35974        144
2:  8997     GSE46416         32</code></pre>
<p>We are left with two datasets. For simplicity, we’ll pick <a href="https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=8997" class="external-link">GSE46416</a> since it has the smaller number of samples. Now that we have the ID for our experiment, we can fetch the data associated with it.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getDatasetsInfo.html">getDatasetsInfo</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Description</span><span class="op">)</span></span></code></pre></div>
<pre><code>   ee.ShortName                                                         ee.Name
1:     GSE46416 State- and trait-specific gene expression in euthymia and mania
   ee.ID
1:  8997
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               ee.Description
1: Gene expression profiles of bipolar disorder (BD) patients were assessed during both a manic and a euthymic phase and compared both intra-individually, and with the gene expression profiles of controls.\nLast Updated (by provider): Sep 05 2014\nContributors:  Christian C Witt Benedikt Brors Dilafruz Juraeva Jens Treutlein Carsten Sticht Stephanie H Witt Jana Strohmaier Helene Dukal Josef Frank Franziska Degenhardt Markus M Nöthen Sven Cichon Maren Lang Marcella Rietschel Sandra Meier Manuel Mattheisen</code></pre>
<p>To access the expression data in a convenient form, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getDataset.html" class="external-link"><code>getDataset()</code></a>. It is a high-level wrapper that combines various endpoint calls to return an annotated <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link"><code>SummarizedExperiment</code></a> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf" class="external-link"><code>ExpressionSet</code></a> of the queried dataset that is compatible with other Bioconductor packages for downstream analyses. They include the expression matrix along with the experimental design, and ensure the sample names match between both when transforming/subsetting data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDataset.html">getDataset</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span></span></code></pre></div>
<p>To show how subsetting works, we’ll keep the manic phase data and the <code>reference_subject_role</code>s, which refers to the control samples in Gemma datasets.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check the levels of the disease factor</span></span>
<span><span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "reference_subject_role"              "euthymic_phase_|_Bipolar_Disorder_|"
[3] "bipolar_disorder_|_manic_phase_|"   </code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset patients during manic phase and controls</span></span>
<span><span class="va">manic</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">[</span>, <span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"bipolar_disorder_|_manic_phase_|"</span> <span class="op">|</span> </span>
<span>        <span class="va">dat</span><span class="op">$</span><span class="va">disease</span> <span class="op">==</span> <span class="st">"reference_subject_role"</span><span class="op">]</span></span>
<span><span class="va">manic</span></span></code></pre></div>
<pre><code>class: SummarizedExperiment 
dim: 21986 21 
metadata(8): title abstract ... GemmaSuitabilityScore taxon
assays(1): counts
rownames(21986): 2315430 2315554 ... 7385683 7385696
rowData names(3): GeneSymbol GeneName NCBIid
colnames(21): Control,1_DE50 Control,12 ...
  Bipolardisorderpatientmanicphase,16
  Bipolardisorderpatientmanicphase,21
colData names(2): batch disease</code></pre>
<p>Let’s check the expression for every sample to make sure they look OK:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get Expression matrix</span></span>
<span><span class="va">manicExpr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">manic</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">manicExpr</span>, pch <span class="op">=</span> <span class="st">"."</span>, xaxt <span class="op">=</span> <span class="st">"n"</span>, xlab <span class="op">=</span> <span class="st">"Sample"</span>, ylab <span class="op">=</span> <span class="st">"Expression"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/boxplot-1.png" alt="Gene expression distributions of bipolar patients during manic phase and controls." width="700"><p class="caption">
Gene expression distributions of bipolar patients during manic phase and controls.
</p>
</div>
<p>You can also use <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetExpression.html" class="external-link"><code>getDatasetExpression()</code></a> to only get the expression matrix, and <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetDesign.html" class="external-link"><code>getDatasetDesign()</code></a> to get the experimental design matrix. Finally, you can use <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetTidy.html" class="external-link"><code>getDatasetTidy()</code></a> to get a tibble that combines the expression and design matrices in <a href="https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html" class="external-link">tidy format</a>. This format is less memory-efficient than <a href="https://pavlidislab.github.io/gemma.R/reference/getDataset.html" class="external-link"><code>getDataset()</code></a>, but allows easy visualization and exploration with <a href="https://ggplot2.tidyverse.org/index.html" class="external-link">ggplot2</a> and the rest of the <a href="https://www.tidyverse.org/" class="external-link">tidyverse</a>.</p>
<div class="section level3">
<h3 id="differential-expression-analyses">Differential expression analyses<a class="anchor" aria-label="anchor" href="#differential-expression-analyses"></a>
</h3>
<p>Gemma also contains precomputed differential expression analyses for most of its datasets, and some datasets contain more than one analysis to account for different factors and their interactions. These tables are stored as resultSets, and you can access them using <a href="https://pavlidislab.github.io/gemma.R/reference/getDatasetDE.html" class="external-link"><code>getDatasetDE()</code></a>. From here on, we can explore and visualize the data to find the most differentially-expressed genes:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getDatasetDE.html">getDatasetDE</a></span><span class="op">(</span><span class="st">"GSE46416"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Classify probes for plotting</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">&lt;-</span> <span class="st">"No"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span> <span class="op">&gt;</span> <span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Up"</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span> <span class="op">&lt;</span> <span class="op">-</span><span class="fl">1.0</span> <span class="op">&amp;</span> </span>
<span>        <span class="va">de</span><span class="op">$</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Down"</span></span>
<span></span>
<span><span class="co"># Upregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Up"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol contrast_bipolar disorder, manic phase_pvalue
 1: 2319550       RBP7                                     0.0001795
 2: 2548699     CYP1B1                                     0.0002182
 3: 3907190       SLPI                                     0.0006255
 4: 3629103      PCLAF                                     0.0009611
 5: 3545525      SLIRP                                     0.0010440
 6: 3146433      COX6C                                     0.0016410
 7: 2538349                                                0.0021300
 8: 2899102       H3C3                                     0.0021460
 9: 3635198     BCL2A1                                     0.0029750
10: 2633191      GPR15                                     0.0039170
    contrast_bipolar disorder, manic phase_logFoldChange
 1:                                                1.074
 2:                                                1.322
 3:                                                1.056
 4:                                                1.278
 5:                                                1.349
 6:                                                1.467
 7:                                                1.073
 8:                                                1.026
 9:                                                1.080
10:                                                1.205</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Downregulated probes</span></span>
<span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">de</span>, <span class="va">diffexpr</span> <span class="op">==</span> <span class="st">"Down"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>, <span class="va">GeneSymbol</span>, <span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span>, </span>
<span>        <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Probe GeneSymbol contrast_bipolar disorder, manic phase_pvalue
 1: 2775390                                                6.260e-06
 2: 3760268                                                3.061e-05
 3: 3124344                                                2.779e-04
 4: 3673179                                                3.131e-04
 5: 3245871      WDFY4                                     3.374e-04
 6: 3022689   SND1-IT1                                     4.476e-04
 7: 2679014                                                5.735e-04
 8: 3336402      RBM14                                     6.753e-04
 9: 4019758                                                6.853e-04
10: 3384417 ANKRD42-DT                                     7.165e-04
    contrast_bipolar disorder, manic phase_logFoldChange
 1:                                               -1.556
 2:                                               -1.851
 3:                                               -1.037
 4:                                               -1.034
 5:                                               -1.157
 6:                                               -1.220
 7:                                               -1.175
 8:                                               -1.071
 9:                                               -1.405
10:                                               -1.003</code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Add gene symbols as labels to DE genes</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span><span class="va">de</span><span class="op">$</span><span class="va">delabel</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">de</span><span class="op">$</span><span class="va">GeneSymbol</span><span class="op">[</span><span class="va">de</span><span class="op">$</span><span class="va">diffexpr</span> <span class="op">!=</span> <span class="st">"No"</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Volcano plot for bipolar patients vs controls</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">de</span>,</span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="va">`contrast_bipolar disorder, manic phase_logFoldChange`</span>,</span>
<span>        y <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">`contrast_bipolar disorder, manic phase_pvalue`</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="va">diffexpr</span>,</span>
<span>        label <span class="op">=</span> <span class="va">delabel</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"gray45"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"log2(FoldChange)"</span>, y <span class="op">=</span> <span class="st">"-log10(p-value)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html" class="external-link">geom_text_repel</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="gemma.R_files/figure-html/diffExpr-1.png" alt="Differentially-expressed genes in bipolar patients during manic phase versus controls." width="700"><p class="caption">
Differentially-expressed genes in bipolar patients during manic phase versus controls.
</p>
</div>
</div>
<div class="section level3">
<h3 id="larger-queries">Larger queries<a class="anchor" aria-label="anchor" href="#larger-queries"></a>
</h3>
<p>The <code>*Info()</code> endpoints accept multiple identifiers in a single function call. For example, getting information on 2 datasets at the same time.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getDatasetsInfo.html">getDatasetsInfo</a></span><span class="op">(</span>datasets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.ShortName</span>, <span class="va">ee.Name</span>, <span class="va">ee.ID</span>, <span class="va">ee.Accession</span>, <span class="va">ee.Samples</span>, <span class="va">taxon.Name</span><span class="op">)</span></span></code></pre></div>
<pre><code>   ee.ShortName                                                         ee.Name
1:     GSE35974                 Expression data from the human cerebellum brain
2:     GSE46416 State- and trait-specific gene expression in euthymia and mania
   ee.ID ee.Accession ee.Samples taxon.Name
1:  5939     GSE35974        144      human
2:  8997     GSE46416         32      human</code></pre>
<p>To query large amounts of data, the API has a pagination system which uses the <code>limit</code> and <code>offset</code> parameters. To avoid overloading the server, calls are limited to a maximum of 100 entries, so the offset allows you to get the next batch of entries in the next call(s). For simplicity, this example shows how pagination works with 5 entries per query.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">10</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">offset</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/getPlatformsInfo.html">getPlatformsInfo</a></span><span class="op">(</span>limit <span class="op">=</span> <span class="fl">5</span>, offset <span class="op">=</span> <span class="va">offset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">platform.ID</span>, <span class="va">platform.ShortName</span>, <span class="va">taxon.Name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code>[[1]]
   platform.ID platform.ShortName taxon.Name
1:           1              GPL96      human
2:           2            GPL1355        rat
3:           3            GPL1261      mouse
4:           4             GPL570      human
5:           5              GPL81      mouse

[[2]]
   platform.ID platform.ShortName taxon.Name
1:           6              GPL85        rat
2:           7             GPL339      mouse
3:           8              GPL91      human
4:           9             GPL890        rat
5:          10            GPL1406      mouse

[[3]]
   platform.ID platform.ShortName taxon.Name
1:          11             GPL891      mouse
2:          12              GPL82      mouse
3:          13             GPL560      mouse
4:          14            GPL1073      mouse
5:          16             GPL369      mouse</code></pre>
<p>The rest of the endpoints only support a single identifier:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE46416"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code>Error: Please specify one valid identifier for dataset.</code></pre>
<p>In these cases, you will have to loop over all the identifiers you wish to query and send separate requests.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GSE35974"</span>, <span class="st">"GSE12649"</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="../reference/getDatasetAnnotations.html">getDatasetAnnotations</a></span><span class="op">(</span><span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>ee.shortName <span class="op">=</span> <span class="va">dataset</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ee.shortName</span>, <span class="va">class.Type</span>, <span class="va">class.Name</span>, <span class="va">term.Name</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code>[[1]]
   ee.shortName    class.Type     class.Name         term.Name
1:     GSE35974   FactorValue        disease  Bipolar Disorder
2:     GSE35974   BioMaterial biological sex              male
3:     GSE35974   BioMaterial biological sex            female
4:     GSE35974   FactorValue        disease     schizophrenia
5:     GSE35974 ExperimentTag  organism part        cerebellum
6:     GSE35974   FactorValue        disease mental depression

[[2]]
   ee.shortName    class.Type    class.Name              term.Name
1:     GSE12649   BioMaterial organism part reference subject role
2:     GSE12649   BioMaterial       disease       Bipolar Disorder
3:     GSE12649 ExperimentTag organism part      prefrontal cortex
4:     GSE12649   BioMaterial       disease          schizophrenia</code></pre>
</div>
</div>
<div class="section level2">
<h2 id="output-options">Output options<a class="anchor" aria-label="anchor" href="#output-options"></a>
</h2>
<div class="section level3">
<h3 id="raw-data">Raw data<a class="anchor" aria-label="anchor" href="#raw-data"></a>
</h3>
<p>By default, Gemma API does some parsing on the raw API results to make it easier to work with inside of R. In the process, it drops some typically unused values. If you wish to fetch everything, use <code>raw = TRUE</code>. Instead of a data table, you’ll usually be served a list that represents the underlying JSON response.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getGeneLocation.html">getGeneLocation</a></span><span class="op">(</span><span class="st">"DYRK1A"</span><span class="op">)</span></span></code></pre></div>
<pre><code>   chromosome strand bin nucleotide length taxon.Name  taxon.Scientific
1:         11      + 105   33890705 118714        rat Rattus norvegicus
2:         21      + 108   37365572 160785      human      Homo sapiens
3:         16      +  20   94370636 125741      mouse      Mus musculus
   taxon.ID taxon.NCBI taxon.Database.Name
1:        3      10116                 rn6
2:        1       9606                hg38
3:        2      10090                mm10</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/getGeneLocation.html">getGeneLocation</a></span><span class="op">(</span><span class="st">"DYRK1A"</span>, raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code>  id nucleotide nucleotideLength strand bin chromosome taxon.id
1 NA   33890705           118714      + 105         11        3
2 NA   37365572           160785      + 108         21        1
3 NA   94370636           125741      +  20         16        2
  taxon.scientificName taxon.commonName taxon.ncbiId taxon.isSpecies
1    Rattus norvegicus              rat        10116              NA
2         Homo sapiens            human         9606              NA
3         Mus musculus            mouse        10090              NA
  taxon.isGenesUsable taxon.externalDatabase.name taxon.externalDatabase.id
1                TRUE                         rn6                        86
2                TRUE                        hg38                        87
3                TRUE                        mm10                        81
  taxon.externalDatabase.checked
1                          FALSE
2                          FALSE
3                          FALSE</code></pre>
</div>
<div class="section level3">
<h3 id="file-outputs">File outputs<a class="anchor" aria-label="anchor" href="#file-outputs"></a>
</h3>
<p>Sometimes, you may wish to save results to a file for future inspection. You can do this simply by providing a filename to the <code>file</code> parameter. The extension for this file will be one of three options:</p>
<ol style="list-style-type: decimal">
<li>
<code>.json</code>, if you requested results with <code>raw=TRUE</code>
</li>
<li>
<code>.csv</code> if the results have no nested data tables</li>
<li>
<code>.rds</code> otherwise</li>
</ol>
<p>You can also specify whether or not the new fetched results are allowed to overwrite an existing file by specifying the <code>overwrite = TRUE</code> parameter.</p>
</div>
<div class="section level3">
<h3 id="memoise-data">Memoise data<a class="anchor" aria-label="anchor" href="#memoise-data"></a>
</h3>
<p>To speed up results, you can remember past results so future queries can proceed virtually instantly. This is enabled through the <a href="https://github.com/r-lib/memoise" class="external-link"><code>memoise</code></a> package. To enable memoisation, simply set <code>memoised = TRUE</code> in the function call whenever you want to refer to the cache, both to save data for future calls and use the saved data for repeated calls.</p>
<p>By default this will create a cache in your local filesystem.</p>
<p>You can see the path to your cache by using <code>rappdirs::user_cache_dir(appname = "gemmaR")</code>. If you want to use a different directory, use <code>options(gemma.cache = 'path')</code>.</p>
<p>If you’re done with your fetching and want to ensure no space is being used for cached results, or if you just want to ensure you’re getting up-to-date data from Gemma, you can clear the cache using <a href="https://pavlidislab.github.io/gemma.R/reference/forgetGemmaMemoised.html" class="external-link"><code>forgetGemmaMemoised()</code></a>.</p>
</div>
<div class="section level3">
<h3 id="changing-defaults">Changing defaults<a class="anchor" aria-label="anchor" href="#changing-defaults"></a>
</h3>
<p>We’ve seen how to change <code>raw = TRUE</code>, <code>overwrite = TRUE</code> and <code>memoised = TRUE</code> in individual function calls. It’s possible that you want to always use the functions these ways without specifying the option every time. You can do this by simply changing the default, which is visible in the function definition. See below for examples.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.memoised <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always refer to cache</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always overwrite when saving files</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>gemma.raw <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># always recieve results as-is from Gemma</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] SummarizedExperiment_1.26.1 Biobase_2.56.0             
 [3] GenomicRanges_1.48.0        GenomeInfoDb_1.32.3        
 [5] IRanges_2.30.1              S4Vectors_0.34.0           
 [7] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
 [9] matrixStats_0.62.0          ggrepel_0.9.1              
[11] ggplot2_3.3.6               dplyr_1.0.9                
[13] gemma.R_0.99.30             BiocStyle_2.24.0           

loaded via a namespace (and not attached):
 [1] httr_1.4.4             sass_0.4.2             tidyr_1.2.0           
 [4] bit64_4.0.5            jsonlite_1.8.0         bslib_0.4.0           
 [7] BiocManager_1.30.18    highr_0.9              GenomeInfoDbData_1.2.8
[10] yaml_2.3.5             pillar_1.8.1           lattice_0.20-45       
[13] glue_1.6.2             digest_0.6.29          XVector_0.36.0        
[16] colorspace_2.0-3       htmltools_0.5.3        Matrix_1.4-1          
[19] pkgconfig_2.0.3        bookdown_0.28          zlibbioc_1.42.0       
[22] purrr_0.3.4            scales_1.2.1           tibble_3.1.8          
[25] farver_2.1.1           generics_0.1.3         ellipsis_0.3.2        
[28] cachem_1.0.6           withr_2.5.0            cli_3.3.0             
[31] magrittr_2.0.3         crayon_1.5.1           memoise_2.0.1         
[34] evaluate_0.16          fs_1.5.2               fansi_1.0.3           
[37] textshaping_0.3.6      tools_4.2.1            data.table_1.14.2     
[40] lifecycle_1.0.1        stringr_1.4.1          munsell_0.5.0         
[43] DelayedArray_0.22.0    compiler_4.2.1         pkgdown_2.0.6         
[46] jquerylib_0.1.4        systemfonts_1.0.4      rlang_1.0.4           
[49] grid_4.2.1             RCurl_1.98-1.8         rappdirs_0.3.3        
[52] bitops_1.0-7           labeling_0.4.2         rmarkdown_2.16        
[55] gtable_0.3.0           curl_4.3.2             R6_2.5.1              
[58] knitr_1.40             fastmap_1.1.0          bit_4.0.4             
[61] utf8_1.2.2             rprojroot_2.0.3        ragg_1.2.2            
[64] desc_1.4.1             stringi_1.7.8          Rcpp_1.0.9            
[67] vctrs_0.4.1            tidyselect_1.1.2       xfun_0.32             </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
