<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A guide to metadata for samples and differential expression analyses • gemma.R</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A guide to metadata for samples and differential expression analyses">
<meta property="og:description" content="gemma.R">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=TODO"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'TODO');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gemma.R</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.1.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gemma.R.html">Accessing curated gene expression data with gemma.R</a>
    </li>
    <li>
      <a href="../articles/metadata.html">A guide to metadata for samples and differential expression analyses</a>
    </li>
    <li>
      <a href="../articles/metanalysis.html">A meta analysis on effects of Parkinson's Disease using Gemma.R</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PavlidisLab/gemma.R/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="metadata_files/kePrint-0.0.1/kePrint.js"></script><link href="metadata_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A guide to metadata for samples and differential
expression analyses</h1>
                        <h4 data-toc-skip class="author">B. Ogan
Mancarci</h4>
            <address class="author_afil">
      Michael Smith Laboratories, University of British Columbia,
Vancouver, Canada<br><small class="dont-index">Source: <a href="https://github.com/PavlidisLab/gemma.R/blob/HEAD/vignettes/metadata.Rmd" class="external-link"><code>vignettes/metadata.Rmd</code></a></small>
      <div class="hidden name"><code>metadata.Rmd</code></div>

    </address>
</div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pavlidislab.github.io/gemma.R/" class="external-link">gemma.R</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The data in Gemma are manually annotated by curators with terms,
often using an ontology term on both dataset and sample level. In
Gemma.R three primary functions allow access to these annotations for a
given dataset.</p>
<ul>
<li><p><code>get_dataset_annotations</code>: This function returns
annotations associated with a dataset. These try to serve as tags
describing the dataset as a whole and they characteristics that samples
within the datasets have while also including some additional
terms.</p></li>
<li><p><code>get_dataset_samples</code>: This function returns samples
and associated annotations related to their experimental groups for an
experiment</p></li>
<li><p><code>get_dataset_differential_expression_analyses</code>: This
function returns information about differential expression analyses
automatically performed by Gemma for a given experiment. Each row of the
output is a contrast where a specific property or an interaction of
properties are described.</p></li>
</ul>
<p>In the examples below we will be referring to <a href="https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=8972" class="external-link">GSE48962</a>
experiment, where striatum and cerebral cortex samples from control mice
and mice belonging to a Huntington model (R6/2) were taken from 8 week
and 12 week old mice.</p>
</div>
<div class="section level2">
<h2 id="dataset-tags">Dataset tags<a class="anchor" aria-label="anchor" href="#dataset-tags"></a>
</h2>
<p>Terms returned via <code>get_dataset_annotations</code> are tags used
to describe a dataset in general terms.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_dataset_annotations.html">get_dataset_annotations</a></span><span class="op">(</span><span class="st">'GSE48962'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">gemma_kable</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
class.name
</th>
<th style="text-align:left;">
class.URI
</th>
<th style="text-align:left;">
term.name
</th>
<th style="text-align:left;">
term.URI
</th>
<th style="text-align:left;">
object.class
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
<td style="text-align:left;">
Overexpression
</td>
<td style="text-align:left;">
<a href="http://gemma.msl" class="external-link uri">http://gemma.msl</a>…/TGEMO_00004
</td>
<td style="text-align:left;">
FactorValue
</td>
</tr>
<tr>
<td style="text-align:left;">
study design
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0001426
</td>
<td style="text-align:left;">
SUBSET
</td>
<td style="text-align:left;">
<a href="http://gemma.msl" class="external-link uri">http://gemma.msl</a>…/TGEMO_00022
</td>
<td style="text-align:left;">
ExperimentTag
</td>
</tr>
<tr>
<td style="text-align:left;">
strain
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0005135
</td>
<td style="text-align:left;">
C57BL/6
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0022397
</td>
<td style="text-align:left;">
ExperimentTag
</td>
</tr>
<tr>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000635
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/UBERON_0002435
</td>
<td style="text-align:left;">
FactorValue
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
<a href="http://purl.org/" class="external-link uri">http://purl.org/</a>../3064
</td>
<td style="text-align:left;">
FactorValue
</td>
</tr>
<tr>
<td style="text-align:left;">
strain
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0005135
</td>
<td style="text-align:left;">
CBA
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0022463
</td>
<td style="text-align:left;">
ExperimentTag
</td>
</tr>
<tr>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000635
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/UBERON_0000956
</td>
<td style="text-align:left;">
FactorValue
</td>
</tr>
</tbody>
</table>
<p>These tags come as a class/term pairs and inherit any terms that is
assigned to any of the samples. Therefore we can see all chemicals and
cell types used in the experiment.</p>
</div>
<div class="section level2">
<h2 id="factor-values">Factor values<a class="anchor" aria-label="anchor" href="#factor-values"></a>
</h2>
<p>Samples and differential expression contrasts in Gemma are annotated
with factor values. These values contain statements that describe these
samples and which samples belong to which experimental in a differential
expression analysis respectively.</p>
<div class="section level3">
<h3 id="sample-factor-values">Sample factor values<a class="anchor" aria-label="anchor" href="#sample-factor-values"></a>
</h3>
<p>In gemma.R these values are stored in nested <code>data.table</code>s
and can be found by accessing the relevant columns of the outputs.
Annotations for samples can be accessed using
<code>get_dataset_samples</code>. <code>sample.factorValues</code>
column contains the relevant information</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_samples.html">get_dataset_samples</a></span><span class="op">(</span><span class="st">'GSE48962'</span><span class="op">)</span></span>
<span><span class="va">samples</span><span class="op">$</span><span class="va">sample.factorValues</span><span class="op">[[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">samples</span><span class="op">$</span><span class="va">sample.name</span> <span class="op">==</span> <span class="st">"TSM490"</span><span class="op">)</span></span>
<span>    <span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
category.URI
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
value.URI
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
predicate.URI
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
object.URI
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
<th style="text-align:left;">
factor.category.URI
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
block
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0005067
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0073_F…
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0073_F…
</td>
<td style="text-align:right;">
163476
</td>
<td style="text-align:right;">
32643
</td>
<td style="text-align:left;">
block
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0005067
</td>
</tr>
<tr>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000724
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:right;">
120179
</td>
<td style="text-align:right;">
20543
</td>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000724
</td>
</tr>
<tr>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000635
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/UBERON_0002435
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:right;">
120172
</td>
<td style="text-align:right;">
20540
</td>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000635
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
<a href="http://purl.org/" class="external-link uri">http://purl.org/</a>../3064
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/GENO_0000222
</td>
<td style="text-align:left;">
CAG repeats
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
<a href="http://purl.org/" class="external-link uri">http://purl.org/</a>../3064
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
<a href="http://purl.obolibrary" class="external-link uri">http://purl.obolibrary</a>…/GENO_0000222
</td>
<td style="text-align:left;">
Overexpression
</td>
<td style="text-align:left;">
<a href="http://gemma.msl" class="external-link uri">http://gemma.msl</a>…/TGEMO_00004
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
<a href="http://www.ebi" class="external-link uri">http://www.ebi</a>…/EFO_0000513
</td>
</tr>
</tbody>
</table>
<p>The example above shows a single factor value object for one sample.
The rows of this <code>data.table</code> are statements that belong to a
factor value. Below each column of this nested table is described. If a
given field is filled by an ontology term, the corresponding URI column
will contain the ontology URI for the field.</p>
<ul>
<li>
<code>category</code>/<code>category.URI</code>: Category of the
individual statement, such as treatment, phenotype or strain</li>
<li>
<code>value</code>/<code>value.URI</code>: The subject of the
statement.</li>
<li>
<code>predicate</code>/<code>predicate.URI</code>: When a subject
alone is not enough to describe all details, a statement can contain a
predicate and an object. The predicate describes the relationship
between the subject of the statement and the object. In the example
above, these are used to denote properties of the human HTT in the mouse
models</li>
<li>
<code>object</code>/<code>object.URI</code>: The object of a
statement is a property further describing it’s value. In this example
these describe the properties of the HTT gene in the mouse model, namely
that it has CAG repeats and it is overexpressed. If the value was a drug
this could be dosage or timepoint.</li>
<li>
<code>summary</code>: A plain text summary of the factorValue.
Different statements will have the same summary if they are part of the
same factor value</li>
<li>
<code>ID</code>: An integer identifier for the specific factor
value. In the example above, the genotype of the mouse is defined as a
single factor value made up of two statements stating the HTT gene has
CAG repeats and that it is overexpressed. This factor value has the ID
of 120175 which is shared by both rows containing the statements
describing it. This ID will repeat for every other patient that has the
same genotype or differential expression results using that factor as a
part of their contrast. For instance we can see which samples that was
subjected to this condition using this ID instead of trying to match the
other columns describing the statements</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">sample.factorValues</span><span class="op">[[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">samples</span><span class="op">$</span><span class="va">sample.name</span> <span class="op">==</span> <span class="st">"TSM490"</span><span class="op">)</span></span>
<span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="st">"HTT [human] huntingtin"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">$</span><span class="va">ID</span><span class="op">}</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unique</span></span>
<span></span>
<span></span>
<span><span class="co"># count how many patients has this phenotype</span></span>
<span><span class="va">samples</span><span class="op">$</span><span class="va">sample.factorValues</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">x</span><span class="op">$</span><span class="va">ID</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">sum</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 12</span></span></code></pre>
<ul>
<li>
<code>factor.ID</code>: An integer identifier for the factor. A
factor holds specific factor values. For the example above whether or
not the mouse is a wild type mouse or if it has a wild type genotype is
stored under the id 20541</li>
</ul>
<p>We can use this to fetch all distinct genotypes</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="va">samples</span><span class="op">$</span><span class="va">sample.factorValues</span><span class="op">[[</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">samples</span><span class="op">$</span><span class="va">sample.name</span> <span class="op">==</span> <span class="st">"TSM490"</span><span class="op">)</span></span>
<span>    <span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">value</span> <span class="op">==</span> <span class="st">"HTT [human] huntingtin"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">$</span><span class="va">factor.ID</span><span class="op">}</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unique</span></span>
<span></span>
<span><span class="va">samples</span><span class="op">$</span><span class="va">sample.factorValues</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>\<span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">factor.ID</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">$</span><span class="va">summary</span><span class="op">}</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unlist</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">unique</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "wild type genotype"                                                             </span></span>
<span><span class="co">## [2] "HTT [human] huntingtin has_genotype CAG repeats and has_genotype Overexpression"</span></span></code></pre>
<p>This shows us the dataset has control mice and Huntington Disease
model mice.. This ID can be used to match the factor between samples and
between samples and differential expression experiments -
<code>factor.category</code>/<code>factor.category.URI</code>: The
category of the whole factor. Usually this is the same with the
<code>category</code> of the statements making up the factor value.
However in cases like the example above, where the value describes a
treatment while the factor overall represents a phenotype, they can
differ.</p>
<p>gemma.R includes a convenience function to create a simplified design
matrix out of these factor values for a given experiment. This will
unpack the nested data.frames and provide a more human readable output,
giving each available factor it’s own column.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/make_design.html">make_design</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span></span>
<span><span class="va">design</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="co"># first column is just a copy of the original factor values</span></span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
organism part
</th>
<th style="text-align:left;">
block
</th>
<th style="text-align:left;">
timepoint
</th>
<th style="text-align:left;">
genotype
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
ESW176
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0071_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
TCW9469
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0053:L…
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
ECM175
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0071_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
</tr>
<tr>
<td style="text-align:left;">
ESW183
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0072_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
ECW178
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0071_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
TSW479
</td>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0073_F…
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
</tbody>
</table>
<p>Using this output, here we look at the sample sizes for different
experimental groups.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">design</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">`organism part`</span>,<span class="va">timepoint</span>,<span class="va">genotype</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>n<span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## `summarise()` has grouped output by 'organism part', 'timepoint'. You can</span></span>
<span><span class="co">## override using the `.groups` argument.</span></span></code></pre>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
organism part
</th>
<th style="text-align:left;">
timepoint
</th>
<th style="text-align:left;">
genotype
</th>
<th style="text-align:right;">
n
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
striatum
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
3
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="differential-expression-analysis-factor-values">Differential expression analysis factor values<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-factor-values"></a>
</h3>
<p>For most experiments it contains, Gemma performs automated
differential expression analyses. The kinds of analyses that will be
performed is informed by the factor values belonging to the samples.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># removing columns containing factor values and URIs for brevity</span></span>
<span><span class="va">remove_columns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'baseline.factors'</span>,<span class="st">'experimental.factors'</span>,<span class="st">'subsetFactor'</span>,<span class="st">'factor.category.URI'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dea</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_dataset_differential_expression_analyses.html">get_dataset_differential_expression_analyses</a></span><span class="op">(</span><span class="st">"GSE48962"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dea</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">remove_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
result.ID
</th>
<th style="text-align:left;">
contrast.ID
</th>
<th style="text-align:right;">
experiment.ID
</th>
<th style="text-align:left;">
factor.category
</th>
<th style="text-align:left;">
factor.ID
</th>
<th style="text-align:left;">
isSubset
</th>
<th style="text-align:right;">
probes.analyzed
</th>
<th style="text-align:right;">
genes.analyzed
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
492856
</td>
<td style="text-align:left;">
120175_120178
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
genotype,timepoint
</td>
<td style="text-align:left;">
20541,20543
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
18969
</td>
<td style="text-align:right;">
18972
</td>
</tr>
<tr>
<td style="text-align:right;">
492855
</td>
<td style="text-align:left;">
120175
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
20541
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
18968
</td>
<td style="text-align:right;">
18971
</td>
</tr>
<tr>
<td style="text-align:right;">
492854
</td>
<td style="text-align:left;">
120178
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
20543
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
18968
</td>
<td style="text-align:right;">
18971
</td>
</tr>
<tr>
<td style="text-align:right;">
492853
</td>
<td style="text-align:left;">
120175_120178
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
genotype,timepoint
</td>
<td style="text-align:left;">
20541,20543
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
20621
</td>
<td style="text-align:right;">
20623
</td>
</tr>
<tr>
<td style="text-align:right;">
492852
</td>
<td style="text-align:left;">
120175
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
20541
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
20621
</td>
<td style="text-align:right;">
20623
</td>
</tr>
<tr>
<td style="text-align:right;">
492851
</td>
<td style="text-align:left;">
120178
</td>
<td style="text-align:right;">
8972
</td>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
20543
</td>
<td style="text-align:left;">
TRUE
</td>
<td style="text-align:right;">
20621
</td>
<td style="text-align:right;">
20623
</td>
</tr>
</tbody>
</table>
<p>The example above shows the differential expression analyses results.
Each row of this data.table represents a differential expression
contrast connected to a fold change and a p value in the output of
<code>get_differential_expression_values</code> function. If we look at
the <code>contrast.ID</code> we will see the factor value identifiers
returned in the <code>ID</code> column of our
<code>sample.factorValues</code>. These represent which factor value is
used as the experimental factor. Note that some rows will have two IDs
appended together. These represent the interaction effects of multiple
factors. For simplicity, we will start from a contrast without an
interaction.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span> <span class="op">&lt;-</span> <span class="va">dea</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="va">factor.category</span> <span class="op">==</span> <span class="st">"genotype"</span> <span class="op">&amp;</span> </span>
<span>            <span class="va">subsetFactor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="st">'value'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">==</span><span class="st">'cerebral cortex'</span><span class="op">}</span> <span class="co"># we will talk about subsets in a moment</span></span>
<span>        <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># removing URIs for brevity</span></span>
<span><span class="va">uri_columns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'category.URI'</span>,</span>
<span>                <span class="st">'object.URI'</span>,</span>
<span>                <span class="st">'value.URI'</span>,</span>
<span>                <span class="st">'predicate.URI'</span>,</span>
<span>                <span class="st">'factor.category.URI'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">contrast</span><span class="op">$</span><span class="va">baseline.factors</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">uri_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
120174
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr></tbody>
</table>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span><span class="op">$</span><span class="va">experimental.factors</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">uri_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
CAG repeats
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
Overexpression
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr>
</tbody>
</table>
<p>Here, we can see the baseline is the wild type mouse, being compared
to the Huntington Disease models</p>
<p>If we examine a factor with interaction, both baseline and
experimental factor value columns will contain two factor values.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span> <span class="op">&lt;-</span> <span class="va">dea</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="va">factor.category</span> <span class="op">==</span> <span class="st">"genotype,timepoint"</span> <span class="op">&amp;</span> </span>
<span>            <span class="va">subsetFactor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="st">'value'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="op">{</span><span class="va">.</span><span class="op">==</span><span class="st">'cerebral cortex'</span><span class="op">}</span> <span class="co"># we're almost there!</span></span>
<span>        <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span><span class="op">$</span><span class="va">baseline.factors</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">uri_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
wild type genotype
</td>
<td style="text-align:right;">
120174
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:right;">
120179
</td>
<td style="text-align:right;">
20543
</td>
<td style="text-align:left;">
timepoint
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span><span class="op">$</span><span class="va">experimental.factors</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">uri_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
CAG repeats
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
genotype
</td>
<td style="text-align:left;">
HTT [human] huntingtin
</td>
<td style="text-align:left;">
has_genotype
</td>
<td style="text-align:left;">
Overexpression
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
<td style="text-align:right;">
120175
</td>
<td style="text-align:right;">
20541
</td>
<td style="text-align:left;">
genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
timepoint
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:right;">
120178
</td>
<td style="text-align:right;">
20543
</td>
<td style="text-align:left;">
timepoint
</td>
</tr>
</tbody>
</table>
<p>A third place that can contain factorValues is the
<code>subsetFactor</code>. Certain differential expression analyses
exclude certain samples based on a given factor. In this example we can
see that this analysis were only performed on samples from the cerebral
cortex.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast</span><span class="op">$</span><span class="va">subsetFactor</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="op">!</span><span class="va">uri_columns</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
category
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
predicate
</th>
<th style="text-align:left;">
object
</th>
<th style="text-align:left;">
summary
</th>
<th style="text-align:right;">
ID
</th>
<th style="text-align:right;">
factor.ID
</th>
<th style="text-align:left;">
factor.category
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
organism part
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:right;">
120173
</td>
<td style="text-align:right;">
20540
</td>
<td style="text-align:left;">
NA
</td>
</tr></tbody>
</table>
<p>The ids of the factor values included in
<code>baseline.factors</code> and <code>experimental.factors</code>
along with <code>subsetFactor</code> can be used to determine which
samples represent a given contrast. For convenience,
<code>get_dataset_object</code> function which is used to compile
metadata and expression data of an experiment in a single object,
includes <code>resultSets</code> and <code>contrasts</code> argument
which will return the data already composed of samples representing a
particular contrast.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/get_dataset_object.html">get_dataset_object</a></span><span class="op">(</span><span class="st">"GSE48962"</span>,resultSets <span class="op">=</span> <span class="va">contrast</span><span class="op">$</span><span class="va">result.ID</span>,contrasts <span class="op">=</span> <span class="va">contrast</span><span class="op">$</span><span class="va">contrast.ID</span>,type <span class="op">=</span> <span class="st">'list'</span><span class="op">)</span></span>
<span><span class="va">obj</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">design</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
organism part
</th>
<th style="text-align:left;">
block
</th>
<th style="text-align:left;">
timepoint
</th>
<th style="text-align:left;">
genotype
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
TCW9469
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0053:L…
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
ECM175
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0071_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
</tr>
<tr>
<td style="text-align:left;">
ECW178
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0071_F…
</td>
<td style="text-align:left;">
8 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
TCW9451
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWI-EAS413_0047:Lane=2
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
TCW9457
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWUSI-EAS1563_0053:L…
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
wild type genotype
</td>
</tr>
<tr>
<td style="text-align:left;">
TCM9450
</td>
<td style="text-align:left;">
cerebral cortex
</td>
<td style="text-align:left;">
Device=HWI-EAS413_0047:Lane=1
</td>
<td style="text-align:left;">
12 weeks
</td>
<td style="text-align:left;">
HTT [human] huntingtin has_…
</td>
</tr>
</tbody>
</table>
<p>We suggested that the <code>contrast.ID</code> of a contrast also
corresponded to a column in the differential expression results,
acquired by <code>get_differential_expression_values</code>. We can use
what we have learned to take a look at the expression of genes at the
top of the phenotype, treatment interaction. Each result.ID returns its
separate table when accessing differential expression values.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dif_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_differential_expression_values.html">get_differential_expression_values</a></span><span class="op">(</span><span class="st">'GSE48962'</span><span class="op">)</span></span>
<span><span class="va">dif_vals</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">contrast</span><span class="op">$</span><span class="va">result.ID</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">head</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:right;">
NCBIid
</th>
<th style="text-align:left;">
GeneSymbol
</th>
<th style="text-align:left;">
GeneName
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
corrected_pvalue
</th>
<th style="text-align:right;">
rank
</th>
<th style="text-align:right;">
contrast_120175_120178_log2fc
</th>
<th style="text-align:right;">
contrast_120175_120178_tstat
</th>
<th style="text-align:right;">
contrast_120175_120178_pvalue
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
21975
</td>
<td style="text-align:right;">
21975
</td>
<td style="text-align:left;">
Top3a
</td>
<td style="text-align:left;">
topoisomerase (DNA) III alpha
</td>
<td style="text-align:right;">
0.5748
</td>
<td style="text-align:right;">
0.8126
</td>
<td style="text-align:right;">
0.7072
</td>
<td style="text-align:right;">
-0.2205
</td>
<td style="text-align:right;">
-0.5848
</td>
<td style="text-align:right;">
0.5748
</td>
</tr>
<tr>
<td style="text-align:right;">
64580
</td>
<td style="text-align:right;">
64580
</td>
<td style="text-align:left;">
Ndst4
</td>
<td style="text-align:left;">
N-deacetylase/N-sulfotransf…
</td>
<td style="text-align:right;">
0.1378
</td>
<td style="text-align:right;">
0.5046
</td>
<td style="text-align:right;">
0.2730
</td>
<td style="text-align:right;">
-3.0116
</td>
<td style="text-align:right;">
-1.6487
</td>
<td style="text-align:right;">
0.1378
</td>
</tr>
<tr>
<td style="text-align:right;">
208994
</td>
<td style="text-align:right;">
208994
</td>
<td style="text-align:left;">
Fam83b
</td>
<td style="text-align:left;">
family with sequence simila…
</td>
<td style="text-align:right;">
0.4930
</td>
<td style="text-align:right;">
0.7655
</td>
<td style="text-align:right;">
0.6440
</td>
<td style="text-align:right;">
-1.4014
</td>
<td style="text-align:right;">
-0.7184
</td>
<td style="text-align:right;">
0.4930
</td>
</tr>
<tr>
<td style="text-align:right;">
22414
</td>
<td style="text-align:right;">
22414
</td>
<td style="text-align:left;">
Wnt2b
</td>
<td style="text-align:left;">
wingless-type MMTV integrat…
</td>
<td style="text-align:right;">
0.4189
</td>
<td style="text-align:right;">
0.7179
</td>
<td style="text-align:right;">
0.5836
</td>
<td style="text-align:right;">
-0.6944
</td>
<td style="text-align:right;">
-0.8521
</td>
<td style="text-align:right;">
0.4189
</td>
</tr>
<tr>
<td style="text-align:right;">
102640604
</td>
<td style="text-align:right;">
102640604
</td>
<td style="text-align:left;">
Gm36629
</td>
<td style="text-align:left;">
predicted gene, 36629
</td>
<td style="text-align:right;">
0.4834
</td>
<td style="text-align:right;">
0.7592
</td>
<td style="text-align:right;">
0.6366
</td>
<td style="text-align:right;">
1.3084
</td>
<td style="text-align:right;">
0.7349
</td>
<td style="text-align:right;">
0.4834
</td>
</tr>
<tr>
<td style="text-align:right;">
74237
</td>
<td style="text-align:right;">
74237
</td>
<td style="text-align:left;">
Tubgcp2
</td>
<td style="text-align:left;">
tubulin, gamma complex comp…
</td>
<td style="text-align:right;">
0.7909
</td>
<td style="text-align:right;">
0.9210
</td>
<td style="text-align:right;">
0.8587
</td>
<td style="text-align:right;">
-0.0923
</td>
<td style="text-align:right;">
-0.2742
</td>
<td style="text-align:right;">
0.7909
</td>
</tr>
</tbody>
</table>
<p>To get the top genes found associated with this interaction we access
the columns with the correct <code>contrast.ID</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># getting the top 10 genes</span></span>
<span><span class="va">top_genes</span> <span class="op">&lt;-</span> <span class="va">dif_vals</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">contrast</span><span class="op">$</span><span class="va">result.ID</span><span class="op">)</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'contrast_'</span>,<span class="va">contrast</span><span class="op">$</span><span class="va">contrast.ID</span>,<span class="st">'_pvalue'</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">GeneSymbol</span><span class="op">!=</span><span class="st">''</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"|"</span>,<span class="va">GeneSymbol</span>,fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co"># remove blank genes or probes with multiple genes</span></span>
<span>    <span class="op">{</span><span class="va">.</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span><span class="op">}</span></span>
<span><span class="va">top_genes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Probe</span>,<span class="va">NCBIid</span>,<span class="va">GeneSymbol</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>     <span class="fu"><a href="../reference/gemma_kable.html">gemma_kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table table" style="margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:right;">
Probe
</th>
<th style="text-align:right;">
NCBIid
</th>
<th style="text-align:left;">
GeneSymbol
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
67993
</td>
<td style="text-align:right;">
67993
</td>
<td style="text-align:left;">
Nudt12
</td>
</tr>
<tr>
<td style="text-align:right;">
100502959
</td>
<td style="text-align:right;">
100502959
</td>
<td style="text-align:left;">
AV051173
</td>
</tr>
<tr>
<td style="text-align:right;">
12153
</td>
<td style="text-align:right;">
12153
</td>
<td style="text-align:left;">
Bmp1
</td>
</tr>
<tr>
<td style="text-align:right;">
58212
</td>
<td style="text-align:right;">
58212
</td>
<td style="text-align:left;">
Srrm3
</td>
</tr>
<tr>
<td style="text-align:right;">
16969
</td>
<td style="text-align:right;">
16969
</td>
<td style="text-align:left;">
Zbtb7a
</td>
</tr>
<tr>
<td style="text-align:right;">
19732
</td>
<td style="text-align:right;">
19732
</td>
<td style="text-align:left;">
Rgl2
</td>
</tr>
<tr>
<td style="text-align:right;">
382051
</td>
<td style="text-align:right;">
382051
</td>
<td style="text-align:left;">
Pdp2
</td>
</tr>
<tr>
<td style="text-align:right;">
319278
</td>
<td style="text-align:right;">
319278
</td>
<td style="text-align:left;">
Shfl
</td>
</tr>
<tr>
<td style="text-align:right;">
11501
</td>
<td style="text-align:right;">
11501
</td>
<td style="text-align:left;">
Adam8
</td>
</tr>
<tr>
<td style="text-align:right;">
101883
</td>
<td style="text-align:right;">
101883
</td>
<td style="text-align:left;">
Igflr1
</td>
</tr>
</tbody>
</table>
<p>We can then use the expression data returned by
<code>get_dataset_object</code> to examine the expression values for
these genes.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp_subset</span><span class="op">&lt;-</span> <span class="va">obj</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">exp</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">Probe</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top_genes</span><span class="op">$</span><span class="va">Probe</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">top_genes</span><span class="op">$</span><span class="va">GeneSymbol</span></span>
<span></span>
<span><span class="co"># ordering design file</span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="va">obj</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">design</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">genotype</span>,<span class="va">timepoint</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># shorten the resistance label a bit</span></span>
<span><span class="va">design</span><span class="op">$</span><span class="va">genotype</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'HTT'</span>,<span class="va">design</span><span class="op">$</span><span class="va">genotype</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">"Huntington Model"</span></span>
<span></span>
<span><span class="va">exp_subset</span><span class="op">[</span>,<span class="va">.SD</span>,.SDcols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">t</span>  <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">scale</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">t</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span>cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,labels_row <span class="op">=</span> <span class="va">genes</span>,</span>
<span>             annotation_col <span class="op">=</span><span class="va">design</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">genotype</span>,<span class="va">timepoint</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="metadata_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] purrr_1.0.2      pheatmap_1.0.12  dplyr_1.1.4      gemma.R_3.1.5   </span></span>
<span><span class="co">## [5] BiocStyle_2.32.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] rappdirs_0.3.3      sass_0.4.9          utf8_1.2.4         </span></span>
<span><span class="co">##  [4] generics_0.1.3      xml2_1.3.6          stringi_1.8.4      </span></span>
<span><span class="co">##  [7] digest_0.6.35       magrittr_2.0.3      timechange_0.3.0   </span></span>
<span><span class="co">## [10] evaluate_0.24.0     grid_4.4.0          RColorBrewer_1.1-3 </span></span>
<span><span class="co">## [13] bookdown_0.39       fastmap_1.2.0       jsonlite_1.8.8     </span></span>
<span><span class="co">## [16] BiocManager_1.30.23 httr_1.4.7          fansi_1.0.6        </span></span>
<span><span class="co">## [19] viridisLite_0.4.2   scales_1.3.0        textshaping_0.4.0  </span></span>
<span><span class="co">## [22] jquerylib_0.1.4     cli_3.6.2           rlang_1.1.4        </span></span>
<span><span class="co">## [25] bit64_4.0.5         munsell_0.5.1       withr_3.0.0        </span></span>
<span><span class="co">## [28] cachem_1.1.0        yaml_2.3.8          tools_4.4.0        </span></span>
<span><span class="co">## [31] memoise_2.0.1       colorspace_2.1-0    kableExtra_1.4.0   </span></span>
<span><span class="co">## [34] assertthat_0.2.1    curl_5.2.1          vctrs_0.6.5        </span></span>
<span><span class="co">## [37] R6_2.5.1            lubridate_1.9.3     lifecycle_1.0.4    </span></span>
<span><span class="co">## [40] stringr_1.5.1       fs_1.6.4            htmlwidgets_1.6.4  </span></span>
<span><span class="co">## [43] bit_4.0.5           ragg_1.3.2          pkgconfig_2.0.3    </span></span>
<span><span class="co">## [46] desc_1.4.3          pkgdown_2.0.9       pillar_1.9.0       </span></span>
<span><span class="co">## [49] bslib_0.7.0         gtable_0.3.5        data.table_1.15.4  </span></span>
<span><span class="co">## [52] glue_1.7.0          systemfonts_1.1.0   highr_0.11         </span></span>
<span><span class="co">## [55] xfun_0.45           tibble_3.2.1        tidyselect_1.2.1   </span></span>
<span><span class="co">## [58] rstudioapi_0.16.0   knitr_1.47          farver_2.1.2       </span></span>
<span><span class="co">## [61] htmltools_0.5.8.1   svglite_2.1.3       rmarkdown_2.27     </span></span>
<span><span class="co">## [64] compiler_4.4.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Javier Castillo-Arnemann, Jordan Sicherman, Ogan Mancarci, Guillaume Poirier-Morency.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
